VideoLocker/src/main/java/org/edx/mobile/loader/CoursesAsyncLoader.java

```diff
@@ -52,6 +45,18 @@ public CoursesAsyncLoader(Context context) {
         return result;
     }
 
+    private List<EnrolledCoursesResponse> getEnrolledCourses() {
+        List<EnrolledCoursesResponse> ret = null;
+
+        try {
+            ret = environment.getServiceManager().getEnrolledCourses(false);
```

Comment:
I am strongly considering wrapping the retrofit api around another
class/interface…one that caches the method calls (instead of the JSON
responses). Thoughts?

On Wed, May 18, 2016 at 11:54 AM, Christopher Lee <notifications@github.com>
wrote:

> In VideoLocker/src/main/java/org/edx/mobile/loader/CoursesAsyncLoader.java
> <https://github.com/edx/edx-app-android/pull/707#discussion_r63730753>:
>
> > @@ -52,6 +45,18 @@ public CoursesAsyncLoader(Context context) {
> >          return result;
> >      }
> >
> > +    private List<EnrolledCoursesResponse> getEnrolledCourses() {
> > +        List<EnrolledCoursesResponse> ret = null;
> > +
> > +        try {
> > +            ret = environment.getServiceManager().getEnrolledCourses(false);
>
> @mdinino <https://github.com/mdinino> and I had an offline discussion.
> Looks like Retrofit 1.9 itself does not have cacheing, but we should be
> able to cache at the okhttp layer. @mdinino <https://github.com/mdinino>
> will look into it more.
>
> Seems like the issue here was that we need to be able to cache the
> response from Retrofit, where we used to be caching the response in the
> deprecated org.apache.http library.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/edx/edx-app-android/pull/707/files/0c45f11a71cef83eb283660b17c3042aa536934f#r63730753>
>



-- 
Regards,

Marc DiNino
Android Engineer
EDX